src/Controls/tests/Xaml.UnitTests/Issues/Maui23711.xaml.cs

```diff
@@ -0,0 +1,66 @@
+using System.Linq;
+using Microsoft.Maui.ApplicationModel;
+using Microsoft.Maui.Controls.Core.UnitTests;
+using Microsoft.Maui.Devices;
+using Microsoft.Maui.Dispatching;
+
+using Microsoft.Maui.UnitTests;
+using NUnit.Framework;
+
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+
+namespace Microsoft.Maui.Controls.Xaml.UnitTests;
+
+[XamlCompilation(XamlCompilationOptions.Skip)]
+public partial class Maui23711 : ContentPage
+{
+    public Maui23711()
+    {
+        InitializeComponent();
+    }
+
+    public Maui23711(bool useCompiledXaml)
+    {
+        //this stub will be replaced at compile time
+    }
+
+    [TestFixture]
+    class Test
+    {
+        MockDeviceInfo mockDeviceInfo;
+
+        [SetUp]
+        public void Setup()
+        {
+            Application.SetCurrentApplication(new MockApplication());
+            DeviceInfo.SetCurrent(mockDeviceInfo = new MockDeviceInfo());
+            DispatcherProvider.SetCurrent(new DispatcherProviderStub());
+        }
+
+
+        [TearDown] public void TearDown()
+        {
+            AppInfo.SetCurrent(null);
+            DeviceInfo.SetCurrent(null);
+        }
+
+        [Test]
+        public void UsesReflectionBasedBindingsWhenCompilationOfBindingsWithSourceIsDisabled([Values(false, true)] bool compileBindingsWithSource)
+        {
+            MockCompiler.Compile(typeof(Maui23711), out MethodDefinition methodDefinition, compileBindingsWithSource: compileBindingsWithSource);
+            Assert.AreEqual(compileBindingsWithSource, ContainsTypedBindingInstantiation(methodDefinition));
+        }
+
+        static bool ContainsTypedBindingInstantiation(MethodDefinition methodDef)
+            => methodDef.Body.Instructions.Any(instruction =>
+                instruction.OpCode == OpCodes.Newobj
+                && instruction.Operand is MethodReference methodRef
+                && methodRef.DeclaringType.Name == "TypedBinding`2");
```

Comment:
Would the `2` change if a new type was introduced? Would this be more future proof?
```suggestion
                && methodRef.DeclaringType.Name.StartsWith("TypedBinding`"));
```

Reply:
I don't see why we would change `TypedBinding` to have a different number of generic arguments. It hasn't changed in a long time and it is (internal) public API. I would personally keep the test as is for now.